components/reporters/vulnerability-logger/internal/reporter/reporter.go

```diff
@@ -0,0 +1,233 @@
+	}
+	var data map[string]interface{}
+	if err := json.Unmarshal(out, &data); err != nil {
+		return errors.Errorf("could not unmarshal json output: %w", err)
+	}
+
+	// Collect all keys (headers)
+	keySet := map[string]struct{}{}
+	for k := range data {
+		keySet[k] = struct{}{}
+	}
+
+	// Sort keys for consistent column order
+	keys := make([]string, 0, len(keySet))
+	for k := range keySet {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	// Prepare table
+	table := tablewriter.NewWriter(os.Stdout)
+	table.Header(keys)
+
+	for _, rowData := range data {
+		table.Append(rowData)
+	}
+
+	table.Render()
+	return nil
+}
+
+func (j vulnReporter) getWhatWeCareAbout(_ context.Context, finding *vf.VulnerabilityFinding) (*smithyFinding, error) {
+	cV := smithyFinding{}
+	cV.Confidence = finding.Finding.GetConfidenceId().String()
+	cV.Enrichments = make([]*vEnrichment, 0)
+	for _, e := range finding.Finding.GetEnrichments() {
+		cV.Enrichments = append(cV.Enrichments, &vEnrichment{
+			Provider:       e.GetProvider(),
+			Value:          e.GetValue(),
+			EnrichmentName: e.GetName(),
+			Type:           e.GetType(),
+		})
+	}
+	cV.Title = finding.Finding.GetFindingInfo().GetTitle()
+	ds := finding.Finding.GetFindingInfo().GetDataSources()
+	for _, datasource := range ds {
+		datas := ocsffindinginfo.DataSource{}
+		if err := protojson.Unmarshal([]byte(datasource), &datas); err != nil {
+			return nil, errors.Errorf("could not json unmarshal finding: %w", err)
+		}
+		cV.Datasources = append(cV.Datasources, &vDatasource{
+			TargetType: datas.GetTargetType().String(),
+			Uri:        datas.GetUri().GetPath(),
+			RepoURL:    datas.GetSourceCodeMetadata().GetRepositoryUrl(),
+			Reference:  datas.GetSourceCodeMetadata().GetReference(),
+			Purl:       datas.GetOciPackageMetadata().GetPackageUrl(),
+			Tag:        datas.GetOciPackageMetadata().GetTag(),
+		})
+	}
+	cV.Description = finding.Finding.GetMessage()
+	cV.Severity = finding.Finding.GetSeverityId().String()
+	cV.Code = make([]vCode, 0)
+	cV.Pkg = make([]vPkg, 0)
+	for _, v := range finding.Finding.Vulnerabilities {
+		for _, ac := range v.AffectedCode {
+			var c vCode
+			if ac.File != nil {
+				c.Path = *ac.File.Path
+			}
+			if ac != nil && ac.StartLine != nil && ac.EndLine != nil {
+				c.LineRange = fmt.Sprintf("%d-%d", *ac.StartLine, *ac.EndLine)
+			}
+			if ac.Remediation != nil {
+				c.Fix = ac.Remediation.GetDesc()
+			}
+			if v.Cwe != nil {
+				c.Cwe = v.Cwe.GetUid()
+			}
+			cV.Code = append(cV.Code, c)
+		}
+		for _, ap := range v.AffectedPackages {
+			var p vPkg
+			if ap.Purl != nil {
+				p.Purl = *ap.Purl
+			}
+			if ap.Path != nil {
+				p.FoundLoc = *ap.Path
+				p.Remediation = ap.Remediation.GetDesc()
+			}
+			if v.Cve != nil {
+				p.Cve = v.Cve.GetUid()
+			}
+			if v.IsExploitAvailable != nil {
+				p.Exploit = *v.IsExploitAvailable
+			}
+			cV.Pkg = append(cV.Pkg, p)
+		}
+		cV.Vendor = v.GetVendorName()
```

Comment:
Is vendor the same as tool name?

Reply:
I think we still need the tool - it should exist as `productUid`.   That is the only thing missing I think

Reply:
done, good catch